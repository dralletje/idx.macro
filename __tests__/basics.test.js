const path = require('path');
const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  pluginName: 'Babel plugin macros',
  snapshot: true,
  tests: withFilename([
    `
      import idx from '../idx.macro'

      idx(global, _ => _.test.lol);
    `,
  ]),
});

/*
 * This adds the filename to each test so you can do require/import relative
 * to this test file.
 */
function withFilename(tests) {
  return tests.map(t => {
    const test = { babelOptions: { filename: __filename } };
    if (typeof t === 'string') {
      test.code = t;
    } else {
      Object.assign(test, t);
    }
    return test;
  });
}
