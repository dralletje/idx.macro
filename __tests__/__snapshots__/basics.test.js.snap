// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Babel plugin macros 1`] = `
"
import idx from '../idx.macro'

idx(global, _ => _.test.lol);

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref;

(_ref = global) != null ? (_ref = _ref.test) != null ? _ref.lol : _ref : _ref;
"
`;

exports[`1. Babel plugin macros v2 1`] = `
"
import idx from '../idx.macro'
import nonexisting from '/notexisting'

idx(nonexisting, _ => _[1][2][3]);

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref;

import nonexisting from '/notexisting';

(_ref = nonexisting) != null ? (_ref = _ref[1]) != null ? (_ref = _ref[2]) != null ? _ref[3] : _ref : _ref : _ref;
"
`;
